p2p/protocol/identify/obsaddr.go

```diff
@@ -382,16 +382,19 @@ func (oas *ObservedAddrManager) recordObservationUnlocked(conn network.Conn, obs
 	observerString := observerGroup(conn.RemoteMultiaddr())
 	localString := string(conn.LocalMultiaddr().Bytes())
 	ob := observation{
-		seenTime:      now,
-		connDirection: conn.Stat().Direction,
+		seenTime: now,
+		inbound:  conn.Stat().Direction == network.DirInbound,
 	}
 
-	observedAddrs := oas.addrs[localString]
 	// check if observed address seen yet, if so, update it
-	for i, previousObserved := range observedAddrs {
-		if previousObserved.addr.Equal(observed) {
-			observedAddrs[i].seenBy[observerString] = ob
-			observedAddrs[i].lastSeen = now
+	for _, observedAddr := range oas.addrs[localString] {
+		if observedAddr.addr.Equal(observed) {
+			// Don't trump an outbound observation with an inbound
+			// one.
+			ob.inbound = ob.inbound || observedAddr.seenBy[observerString].inbound
+
+			observedAddr.seenBy[observerString] = ob
```

Comment:
Yes, this is intended. I'll leave a comment. Basically, these are independent:

* `inbound`: there was an inbound observation for this address. That is, someone who managed to dial us told us it was good.
* `seenTime`: last time it was observed to avoid garbage collecting.